stream-chat-android-offline/src/test/java/io/getstream/chat/android/offline/querychannels/WhenQuery.kt

```diff
@@ -1,44 +1,242 @@
+                .get()
+
+            sut.query()
+
+            Truth.assertThat(sut.channels.value).isEqualTo(listOf(networkChannel2, networkChannel1))
+        }
+
+    private class Fixture {
+        private var chatClient: ChatClient = mock()
+        private var repositories: RepositoryFacade = mock()
+        private var scope: CoroutineScope = TestCoroutineScope()
+        private var chatDomainImpl: ChatDomainImpl = mock()
+        private var querySort: QuerySort<Channel> = QuerySort()
+
+        fun givenQuerySort(querySort: QuerySort<Channel>) = apply {
+            this.querySort = querySort
+        }
+
+        fun givenRepoFacade(repositoryFacade: RepositoryFacade) = apply {
+            repositories = repositoryFacade
+        }
+
+        fun givenChatDomain(chatDomainImpl: ChatDomainImpl) = apply {
+            this.chatDomainImpl = chatDomainImpl
+        }
+
+        fun givenChatClient(chatClient: ChatClient) = apply {
+            this.chatClient = chatClient
+        }
+
+        fun givenFailedNetworkRequest() = apply {
+            whenever(chatClient.queryChannels(any())) doReturn TestCall(Result(mock()))
+        }
+
+        suspend fun givenQueryChannelsSpec(queryChannelsSpec: QueryChannelsSpec) = apply {
+            whenever(repositories.selectById(any())) doReturn queryChannelsSpec
+            whenever(chatDomainImpl.selectAndEnrichChannels(any(), any())) doReturn emptyList()
+        }
+
+        suspend fun givenDBChannels(dbChannels: List<Channel>) = apply {
+            whenever(chatDomainImpl.channel(any<String>())) doAnswer { invocationOnMock ->
+                val cid = invocationOnMock.arguments[0] as String
+                mock<ChannelControllerImpl> {
+                    on { toChannel() } doReturn dbChannels.first { it.cid == cid }
+                }
+            }
+            whenever(chatDomainImpl.channel(any<Channel>())) doAnswer { invocationOnMock ->
+                val channel = invocationOnMock.arguments[0] as Channel
+                mock<ChannelControllerImpl> {
+                    on { toChannel() } doReturn channel
+                }
+            }
+            whenever(chatDomainImpl.selectAndEnrichChannels(any(), any())) doReturn dbChannels
+        }
+
+        fun givenNetworkChannels(channels: List<Channel>) = apply {
+            whenever(chatClient.queryChannels(any())) doReturn channels.asCall()
+
+            whenever(chatDomainImpl.channel(any<String>())) doAnswer { invocationOnMock ->
+                val cid = invocationOnMock.arguments[0] as String
+                mock<ChannelControllerImpl> {
+                    on { toChannel() } doReturn channels.first { it.cid == cid }
+                }
+            }
+            whenever(chatDomainImpl.channel(any<Channel>())) doAnswer { invocationOnMock ->
+                val channel = invocationOnMock.arguments[0] as Channel
+                mock<ChannelControllerImpl> {
```

Comment:
no need for the `<ChannelControllerImpl>`. Smartcast will figure it out already. Happens in other places.

Reply:
I leave it only for readability, seems it wouldn't be so easy to understand if I did omit here